### PR TITLE
adding support for multiple objects in a file

### LINKED ISSUE #13: Fails with multiple documents
If a yaml file contains multiple definitions the code errors.

Example:
```
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1alpha1
metadata:
  name: readonly-binding
subjects:
  - kind: Group
    name: readonly
roleRef:
  kind: ClusterRole
  name: readonly-role
  apiVersion: rbac.authorization.k8s.io/v1alpha1
---
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1alpha1
metadata:
  name: system-admin-binding
subjects:
  - kind: ServiceAccount
    name: default
    namespace: system
roleRef:
  kind: ClusterRole
  name: admin-role
  apiVersion: rbac.authorization.k8s.io/v1alpha1
```

```
Traceback (most recent call last):
  File "kubediff/kubediff", line 40, in <module>
    main()
  File "kubediff/kubediff", line 34, in main
    failed = check_files(args, printer, options.kubeconfig)
  File "kubediff/kubedifflib/_diff.py", line 137, in check_files
    differences += check_file(printer, path, kubeconfig=kubeconfig)
  File "kubediff/kubedifflib/_diff.py", line 80, in check_file
    expected = yaml.load(stream)
  File "/usr/local/lib/python2.7/site-packages/yaml/__init__.py", line 71, in load
    return loader.get_single_data()
  File "/usr/local/lib/python2.7/site-packages/yaml/constructor.py", line 37, in get_single_data
    node = self.get_single_node()
  File "/usr/local/lib/python2.7/site-packages/yaml/composer.py", line 43, in get_single_node
    event.start_mark)
yaml.composer.ComposerError: expected a single document in the stream
  in "workspace/charts/x/manifests/x.yaml", line 1, column 1
but found another document
  in "workspace/charts/x/manifests/x.yaml", line 10, column 1
```

### PR DESCRIPTION
**Pull request by sstarcher**
Resolves #13

### CODE REVIEW

File: kubedifflib/_diff.py [hunk 1]
```diff
@@ -77,22 +77,24 @@ def check_file(printer, path, kubeconfig=None):
   :return: Number of differences found.
   """
   with open(path, 'r') as stream:
-    expected = yaml.load(stream)
+    expected = yaml.load_all(stream)
 
-  kube_obj = KubeObject.from_dict(expected)
+    differences = 0
```
Comment (tomwilkie) [inline kubedifflib/_diff.py]: Does all the below need indenting under the `with` block?  It seems like everything from `differences = 0` and down should be unindented by 2 spaces.

File: kubedifflib/_diff.py [hunk 2]
```diff
@@ -77,22 +77,24 @@ def check_file(printer, path, kubeconfig=None):
...
-  try:
-    running = kube_obj.get_from_cluster(kubeconfig=kubeconfig)
-  except subprocess.CalledProcessError, e:
-    printer.diff(path, Difference(e.output, None))
-    return 1
+      try:
+        running = kube_obj.get_from_cluster(kubeconfig=kubeconfig)
+      except subprocess.CalledProcessError, e:
+        printer.diff(path, Difference(e.output, None))
+        return 1
```
Comment (tomwilkie) [inline kubedifflib/_diff.py]: Now that this is within a loop, `return 1` is not appropriate - it should be `differences += 1` and `continue`

**Comment by sstarcher** [comment]
I fixed the return 1 issue.  I moved it under the open block as load_all is lazy and does not actually load the file until you attempt to access it and will throw a file closed exception.

**Comment by tomwilkie** [comment]
> I fixed the return 1 issue. I moved it under the open block as load_all is lazy and does not actually load the file until you attempt to access it and will throw a file closed exception.

Ah I see.  Thanks!

**Comment by tomwilkie** [comment]
> I fixed the return 1 issue. I moved it under the open block as load_all is lazy and does not actually load the file until you attempt to access it and will throw a file closed exception.

Ah I see.  Thanks!

**Pull request merged**